src/main/java/org/folio/service/pieces/flows/create/PieceCreateFlowInventoryManager.java

```diff
@@ -29,80 +30,53 @@ public PieceCreateFlowInventoryManager(TitlesService titlesService,
     this.inventoryHoldingManager = inventoryHoldingManager;
   }
 
-  public Future<Void> processInventory(CompositePoLine compPOL,  Piece piece,  boolean createItem, RequestContext requestContext) {
-    if (Boolean.TRUE.equals(compPOL.getIsPackage())) {
-      return packagePoLineUpdateInventory(compPOL, piece, createItem, requestContext);
-    }
-    else
-    {
-       return nonPackagePoLineUpdateInventory(compPOL, piece, createItem, requestContext);
-    }
+  public Future<Void> processInventory(CompositePoLine compPOL, Piece piece, boolean createItem, RequestContext requestContext) {
+    var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, piece.getReceivingTenantId());
+    return updateInventoryForPoLine(compPOL, piece, locationContext, requestContext)
+      .compose(instanceId -> handleHolding(compPOL, piece, instanceId, locationContext))
+      .compose(holdingId -> handleItem(compPOL, createItem, piece, locationContext))
+      .map(itemId -> Optional.ofNullable(itemId).map(piece::withItemId))
+      .mapEmpty();
   }
 
-  private Future<Void> nonPackagePoLineUpdateInventory(CompositePoLine compPOL, Piece piece, boolean createItem, RequestContext requestContext) {
-    return nonPackageUpdateTitleWithInstance(compPOL, piece.getTitleId(), requestContext)
-      .compose(instanceId -> handleHolding(compPOL, piece, instanceId, requestContext))
-      .compose(holdingId -> handleItem(compPOL, createItem, piece, requestContext))
-      .map(itemId -> {
-        Optional.ofNullable(itemId).ifPresent(piece::withItemId);
-        return null;
-      })
-      .mapEmpty();
+  private Future<String> updateInventoryForPoLine(CompositePoLine compPOL, Piece piece, RequestContext locationContext, RequestContext requestContext) {
```

Comment:
Inventory it is very wide word, can be more speocifc updateInventoryInstanceForPoLine

Reply:
Renamed